h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -788,8 +790,22 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
           Log.info("Key " + key + " will be parsed using method " + pm + ".");
 
           if(pm == ParserInfo.ParseMethod.DistributedParse) {
-            new DistributedParse(_vg, localSetup, _vecIdStart, chunkStartIdx, this, key, vec.nChunks()).dfork(vec).getResult(false);
-            for( int i = 0; i < vec.nChunks(); ++i )
+
+            final int[] partitionByIndices;
+            if (_parseSetup._partitionBy != null) {
+              partitionByIndices = new int[_parseSetup._partitionBy.length];
+              for (int i = 0; i < _parseSetup._partitionBy.length; i++) {
+                
+                partitionByIndices[i] = org.apache.commons.lang.ArrayUtils.indexOf(_parseSetup._column_names, _parseSetup._partitionBy[i]);
```

Comment:
why fqn?

Reply:
Because H2O ArrayUtils are used in this class as well. And H2O's indexOf alternative doesn't compare strings well. Better to use the time-proven option.

Reply:
Not user anymore.